frontend/src/feed/PaginationControls.tsx

```diff
@@ -0,0 +1,75 @@
+import { useSearchParams } from "react-router-dom";
+interface PaginationProps {
+  totalPages: number;
+}
+const PaginationControls: React.FC<PaginationProps> = ({ totalPages }) => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const getPage = () => {
+    if (!searchParams) console.error("No search params!");
+    return parseInt(searchParams.get("page") ?? "0");
+  };
+
+  const setPage = (page: number) => {
+    if (!searchParams) console.error("No search params!");
+    setSearchParams({ page: page.toString() });
+  };
+
+  const checkIfLastPage = () => {
+    return getPage() < totalPages - 1 || totalPages === 0;
```

Comment:
Last page is true if current page is lower than totalPages-1? Shouldn't it be getPage() === totalPages - 1?